### PR TITLE
Fix random behaviour on FreeBSD 12 when using Rails with Spring

### PR DESCRIPTION
**Pull request by meanphil**
In FreeBSD 12+ when trying to boot Rails using Spring, Rails tries to use rb-kqueue to add a listener to every file in the project as well as everything under `node_modules/` - this can be thousands of files and will randomly die with:

```
/usr/home/pmurray/rb-kqueue/lib/rb-kqueue.rb:27:in `handle_error': Bad address - KQueue failed: There was an error reading or writing the kevent structure. (Errno::EFAULT)
```

I noticed the `kevent` structure had changed with a new member `ext u_int64_t[4]` added to the end of it, this doesn't seem like it should fix an intermittent error... but it seems to? So I think it might actually be a bug in FreeBSD. But anyway, here's a test case without my fix:

```ruby
def run(queue)
  max = 0
  begin
    2_000.times { 
      queue.watch_file('/usr/home/pmurray/somedir', *[:delete, :write, :extend, :attrib, :rename]) { puts "Something happened" }
      max += 1 
    }
  rescue => e
    puts "Got to #{max}"
    raise e
  end

  puts "Completed #{max}"
end

queue = KQueue::Queue.new
run(queue)
```

Returns 

```
Got to 1671
 /usr/home/pmurray/rb-kqueue/lib/rb-kqueue.rb:27:in `handle_error': Bad address - KQueue failed: There was an error reading or writing the kevent structure. (Errno::EFAULT)
```

Repeated runs ended in random number of successful watches each time, from 83 to well into the thousands. It doesn't matter if you're watching the same file or different files (or directories).

With my fix applied, I can repeatedly, successfully, and reliably make it through all 2000 attempts:

```irb(main):054:0> run(queue)
Completed 2000
=> nil
irb(main):055:0> run(queue)
Completed 2000
=> nil
irb(main):056:0> run(queue)
Completed 2000
=> nil
irb(main):057:0> run(queue)
Completed 2000
=> nil
```

### CODE REVIEW

File: lib/rb-kqueue/native.rb
Comment (gregschmit) [inline lib/rb-kqueue/native.rb:30]: This is the only line I don't fully understand. While I'm a maintainer, I'm not the original author so I'm a little more conservative in what I merge and release; I want to verify that `[:u_int64_t, 4]` is correct here and won't cause regressions, so I'll do some looking to see if I can prove that to myself. Otherwise, everything here makes sense to me.

**Review by gregschmit — APPROVED** [review]

**Comment by gregschmit** [comment]
I approved it so @mat813 can merge if he wants, otherwise I just want to check that one line and maybe do a quick regression test on some old FreeBSD VMs before I'm comfortable merging/releasing.

File: lib/rb-kqueue/native.rb
Comment (meanphil) [reply lib/rb-kqueue/native.rb:30]: If it helps, this is the structure from the queue(2) man page:

```
The kevent structure is defined as:

struct kevent {
        uintptr_t  ident;       /* identifier for this event */
        short     filter;       /* filter for event */
        u_short   flags;        /* action flags for kqueue */
        u_int     fflags;       /* filter flag value */
        int64_t   data;         /* filter data value */
        void      *udata;       /* opaque user data identifier */
        uint64_t  ext[4];       /* extensions */
};
```

So the last member of the struct is an array of 4x uint64_t unsigned 64bit integers. Actually, just noticed one of the other struct members has changed as well...

**Comment by mat813** [comment]
Well, to be honest, I am not the original author either, I took over maintainership because I used to use this a decade ago, and I had managed to coerce it into working again.

**Review by mat813 — COMMENTED** [review]
Well, to be honest, I am not the original author either, I took over maintainership because it was abandonned, and I used to use this like a decade and a half ago, and I had managed to coerce it into working again.

**Comment by gregschmit** [comment]
Ok, well it looks good to me; I can probably merge/release later this evening.

**Pull request merged**
